File: BaseMvp/src/main/java/com/base/applemvp/factroy/IMvpPresenterFactroy.java
package com.base.applemvp.factroy;


import java.util.List;

/**
 * applehsp
 * <p>
 * 创建Presenter工厂接口
 */

public interface IMvpPresenterFactroy<P> {
    /**
     * 创建Presenter的接口方法
     *
     * @return 需要创建的Presenter
     */
    List<P> createMvpPresenter();
}

File: BaseMvp/src/main/java/com/base/applemvp/factroy/IMvpPresenterFactroy.kt
package com.base.applemvp.factroy

import java.util.ArrayList

/**
 * applehsp
 *
 *
 * 创建Presenter工厂接口
 */
interface IMvpPresenterFactroy<P> {
    /**
     * 创建Presenter的接口方法
     *
     * @return 需要创建的Presenter
     */
    fun createMvpPresenter(): ArrayList<P>?
}